### PR TITLE
[bluebird-global] Fix finally

### PR DESCRIPTION
**Pull request by alecmev**
Needed for TS 2.7. Not sure if the right approach, but it works.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by d-ph** [comment]
Hi Olegs,

Could you confirm the problem occurs only, when you have `esnext` listed in `lib` in your `tsconfig.json`, please? I wasn't able to reproduce the problem in any other way.

Could you also merge my changes to your branch, please? [link](https://github.com/d-ph/DefinitelyTyped/tree/patch-1). In the diff you'll see a slight change to the line you added and also some unrelated docs changes (let's just do it in one PR).

Btw. I'm not very happy to sort of "rely" on typings from `esnext.d.ts`, which have no guarantees on the stability. But I'm ok to merge this change, since I don't expect for this to change too much.

**Comment by jeremejevs** [comment]
Hi @d-ph,

Yes, this is specific to the new ES2018/ESNext. Reverting the change and targeting ES2017 yields no errors.

Thanks for the corrections, merged and pushed. However, please, take a look at the build log, something appears to be wrong.

About the reliance on `lib.esnext` - sorry, got no input on this. I'm not familiar with the typing guidelines / best practices, and this particular package isn't even an ordinary case, being an augmentation of the standard library.

**Comment by d-ph** [comment]
That's all fine.

Could you revert the `tsconfig.json` line back to say "es2017" instead of "es2018", please? Sorry, that was my bad.

Anyway, thanks for this PR in general.

**Comment by jeremejevs** [comment]
@d-ph No worries, done.

**Comment by d-ph** [comment]
Nice.

One last thing: Are you able to set me as a reviewer of this PR? If yes, then please do and I will officially approve this PR, which might speed up the merging of it. Otherwise let's wait a week to let it follow the normal process.

**Comment by jeremejevs** [comment]
Doesn't seem like I am [able to], unfortunately.

**Comment by d-ph** [comment]
Ok.

Let's just sit back and wait, then.

Thanks again for the contribution.

**Comment by jeremejevs** [comment]
Thanks for the help!

File: types/bluebird-global/index.d.ts [hunk 1]
```diff
@@ -65,7 +81,7 @@ declare global {
         each: typeof Bluebird.prototype.each;
         error: typeof Bluebird.prototype.error;
         filter: typeof Bluebird.prototype.filter;
-        finally: typeof Bluebird.prototype.finally;
+        // finally: typeof Bluebird.prototype.finally;
```
Comment (DanielRosenwasser) [inline types/bluebird-global/index.d.ts]: Remove this comment.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by DanielRosenwasser** [comment]
@rbuckton would you mind taking a quick peek here?

File: types/bluebird-global/index.d.ts (hunk 1 above)
Comment (jeremejevs) [reply types/bluebird-global/index.d.ts]: I have left it in intentionally, as to be in-line with `catch` and `then` (which, I assume, are hints that their definitions are being overridden below).

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**